crates/bevy_reflect/src/serde/ser.rs

```diff
@@ -185,35 +237,31 @@ impl<'a> Serialize for TupleStructSerializer<'a> {
     where
         S: serde::Serializer,
     {
-        let mut state = serializer.serialize_map(Some(2))?;
-
-        state.serialize_entry(type_fields::TYPE, self.tuple_struct.type_name())?;
-        state.serialize_entry(
-            type_fields::TUPLE_STRUCT,
-            &TupleStructValueSerializer {
-                tuple_struct: self.tuple_struct,
-                registry: self.registry,
-            },
+        let type_info = get_type_info(
+            self.tuple_struct.get_type_info(),
+            self.tuple_struct.type_name(),
+            self.registry,
         )?;
-        state.end()
-    }
-}
 
-pub struct TupleStructValueSerializer<'a> {
-    pub tuple_struct: &'a dyn TupleStruct,
-    pub registry: &'a TypeRegistry,
-}
+        let tuple_struct_info = match type_info {
+            TypeInfo::TupleStruct(tuple_struct_info) => tuple_struct_info,
+            info => {
+                return Err(Error::custom(format_args!(
+                    "expected tuple struct type but received {:?}",
+                    info
+                )));
+            }
+        };
 
-impl<'a> Serialize for TupleStructValueSerializer<'a> {
-    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: serde::Serializer,
-    {
-        let mut state = serializer.serialize_seq(Some(self.tuple_struct.field_len()))?;
         let serialization_data = self
             .registry
-            .get_with_name(self.tuple_struct.type_name())
+            .get(self.tuple_struct.type_id())
```

Comment:
Is there any performance difference between lookup of `TypeId` vs `String`?

Reply:
Yup. We should prefer type_id() whenever possible. Getting with type_name will do a hash of the string.

Reply:
Sounds good! I'll update the other instance (which is using `type_name`)